packages/ibm-products/src/components/TagOverflow/TagOverflowModal.js

```diff
@@ -0,0 +1,134 @@
+//
+// Copyright IBM Corp. 2024, 2024
+//
+// This source code is licensed under the Apache-2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+//
+
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+
+import {
+  ComposedModal,
+  ModalHeader,
+  ModalBody,
+  Search,
+  Tag,
+} from '@carbon/react';
+
+import { pkg } from '../../settings';
+import { prepareProps } from '../../global/js/utils/props-helper';
+import { usePortalTarget } from '../../global/js/hooks/usePortalTarget';
+
+const componentName = 'TagOverflowModal';
+const blockClass = `${pkg.prefix}--tag-overflow-modal`;
+
+// Default values for props
+const defaults = {
+  // marked as required by TagSet if needed, default used to satisfy <Search /> component
+  searchLabel: '',
+};
+
+export const TagOverflowModal = ({
+  // The component props, in alphabetical order (for consistency).
+
+  allTags,
+  className,
+  title,
+  onClose,
+  open,
+  portalTarget: portalTargetIn,
+  searchLabel = defaults.searchLabel,
+  searchPlaceholder,
+
+  // Collect any other property values passed in.
+  ...rest
+}) => {
+  const [filteredModalTags, setFilteredModalTags] = useState([]);
+  const [search, setSearch] = useState('');
+  const renderPortalUse = usePortalTarget(portalTargetIn);
+
+  useEffect(() => {
```

Comment:
the code here looks good, but you don't actually need to use an effect here. since the search isn't coming from another source you can just convert this to be a getter function. something like

```
const getFilteredItems = () => {
...
```

and then you could either call `getFilteredItems().map` in the return or set a variable `const filteredItems = getFilteredItems()`

Reply:
Done